CMakeLists.txt

```diff
@@ -64,26 +64,34 @@ if (NOT DEFINED USE_PYTHON_CASADI)
   set(USE_PYTHON_CASADI TRUE)
 endif()
 
+# Use importlib to find the casadi path without importing it. This is useful
+# to find the path for the build-time dependency, not the run-time dependency.
 execute_process(
     COMMAND "${PYTHON_EXECUTABLE}" -c
-    "import casadi as _; print(_.__path__[0])"
+    "import importlib.util; print(importlib.util.find_spec('casadi').submodule_search_locations[0])"
```

Comment:
Changing this command also fixed the error while building the manylinux2014 aarch64 wheel on a previous run under QEMU emulation: https://github.com/agriyakhetarpal/PyBaMM/actions/runs/7006178787

but the wheel has other issues, sadly, with a new unresolved symbol at the time of import...